src/FtpExtractor.php

```diff
@@ -50,6 +51,21 @@ public function __construct(bool $onlyNewFiles, FtpFilesystem $ftpFs, LoggerInte
 
     public function copyFiles(string $sourcePath, string $destinationPath, FileStateRegistry $registry): int
     {
+        try {
+            /** @var AbstractFtpAdapter $adapter */
+            $adapter = $this->ftpFilesystem->getAdapter();
+            $adapter->getConnection();
+            $this->logger->info("Connected to host");
+        } catch (\RuntimeException $e) {
+            throw new UserException($e->getMessage(), $e->getCode(), $e);
+        } catch (\LogicException $e) {
+            throw new UserException($e->getMessage(), $e->getCode(), $e);
```

Comment:
a konvertovat logicException na user exception?

i runtime mi prijde dost na hrane

Reply:
Nj, ono to bohužel vyhazuje tyhle uvnitř, myslím. Každopádně se to takhle chovalo i předtím.

Reply:
tomu rozumim ze je to vyhazuje uvnitr, ale proc LogicException konvertovat na userException? ma to byt celkem jednoznacne appexception, ne?

ale tomu hlavne nerozumim jak se to mohlo takhle chovat predtim, kdyz tam tahle konverze nebyla

Reply:
https://github.com/keboola/ex-ftp/blob/master/src/FtpExtractor.php#L77 (master)

Reply:
A ty logic exceptiony jsou z ftps adapteru, který je IMHO debilně napsaný a měl by tady spíš vyhazovat něco lepšího.

Reply:
https://github.com/thephpleague/flysystem-sftp/blob/master/src/SftpAdapter.php#L193

Reply:
Jakože "invalid username" není LogicException, ale spíš InvalidArgumentException nebo tak něco...

Reply:
Takový to 
![image](https://user-images.githubusercontent.com/642928/60595662-b8f9f900-9da7-11e9-957a-d1159c543619.png)

Reply:
nojo uz to chapu, no to by bylo nejlepsi ty catche vyhodit odsud https://github.com/keboola/ex-ftp/blob/781b6ab61c797c7db90ffc5f91041882d0bcc2d1/src/FtpExtractor.php#L104

takhle je to teda takovy wtf, ale sere pes

Reply:
No, to nejde. Ono se to může odpojit  a tím pádem když to kdekoli zavolá getConnection(), tak to udělá ten disconnect+connect a může to vyhodit tu exception :( Ale je pravda, že když to projde tím přípojením na začátku, tak to bude vyhazovat jiné než tu logic asi...

Reply:
`class ExceptionConverter` ?

Reply:
No to by asi bylo nejlepší řešení. Ale už je to mergnuté, tak přidám jen issue.

Reply:
https://github.com/keboola/ex-ftp/issues/21